rclcpp/src/rclcpp/experimental/executors/events_executor/events_executor.cpp

```diff
@@ -61,6 +64,10 @@ EventsExecutor::EventsExecutor(
       this->refresh_current_collection_from_callback_groups();
     });
 
+  // Make sure that the notify waitable is immediately added to the collection
+  // to avoid missing events
+  this->add_notify_waitable_to_collection(current_entities_collection_->waitables);
```

Comment:
This is the fix: we immediately add the notify waitable to the current collection in the executor's constructor